src/unity/toolkits/object_detection/one_shot_object_detection/util/superposition.cpp

```diff
@@ -0,0 +1,128 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+ */
+#include <boost/gil/extension/numeric/sampler.hpp>
+#include <boost/gil/extension/numeric/resample.hpp>
+
+#include <unity/toolkits/object_detection/one_shot_object_detection/util/mapping_function.hpp>
+#include <unity/toolkits/object_detection/one_shot_object_detection/util/superposition.hpp>
+#include <unity/toolkits/object_detection/one_shot_object_detection/util/quadrilateral_geometry.hpp>
+
+#define BLACK boost::gil::rgb8_pixel_t(0,0,0)
+#define WHITE boost::gil::rgb8_pixel_t(255,255,255)
+#define RGBA_WHITE boost::gil::rgba8_pixel_t(255,255,255,0)
+
+namespace turi {
+namespace one_shot_object_detection {
+
+namespace data_augmentation {
+
+namespace superposition {
```

Comment:
I don't think we need an entire namespace just for this file. As a rule of thumb, aligning namespaces with directories tends to be the cleanest and most intuitive

Reply:
Fixed.